IndoorMapsBackend/src/resolvers/NavResolver.ts

```diff
@@ -62,77 +74,89 @@ export class NavResolver {
-            if(insideArea) {
-                startIgnorableWalls.push(...insideArea.geometry.coordinates[0].map(position => new LatLng(position[1], position[0])).map((latLng, i, arr) => new Wall(latLng, arr[(i + 1) % arr.length])))
+            const fromShape = toArea.floor.areas.find((area) => pointInPolygon(fromLatLon!, (area.shape as unknown as GeoJSON.Feature<GeoJSON.Polygon>).geometry.coordinates[0]))?.shape as unknown | undefined as GeoJSON.Feature<GeoJSON.Polygon> | undefined;
+            if (fromShape) {
+                fromAreaIgnorableWalls.push(...fromShape.geometry.coordinates[0].map(position => new LatLng(position[1], position[0])).map((latLng, i, arr) => new Wall(latLng, arr[(i + 1) % arr.length])))
             }
         }
         else {
-            const fromArea = await ctx.prisma.area.findUnique({
+            fromArea = await ctx.prisma.area.findUnique({
                 where: {
                     id: data.areaFromId,
                 }
             })
             if (!fromArea || !fromArea.shape) throw throwGraphQLBadInput('From Area not found')
             if (fromArea.floorId !== toArea.floorId) throw throwGraphQLBadInput("Navigation between floors is not currently supported")
             const fromShape = fromArea.shape as unknown as GeoJSON.Feature<GeoJSON.Polygon>
-            startIgnorableWalls.push(...fromShape.geometry.coordinates[0].map(position => new LatLng(position[1], position[0])).map((latLng, i, arr) => new Wall(latLng, arr[(i + 1) % arr.length])))
+            fromAreaIgnorableWalls.push(...fromShape.geometry.coordinates[0].map(position => new LatLng(position[1], position[0])).map((latLng, i, arr) => new Wall(latLng, arr[(i + 1) % arr.length])))
             fromLatLon = findPolygonCenter(fromShape)
         }
 
         const toLatLon = findPolygonCenter(toShape)
 
         if (!toLatLon || !fromLatLon) throw throwGraphQLBadInput('Issue with starting or ending GPS locations')
 
+        const pathfindingMethod = data.pathfindingMethod ?? PathfindingMethod.Standard;
         let navMesh: NavMesh;
         let walls: Wall[] = [];
         let neededToGenerateANavMesh = false
 
-        if (toArea.floor.navMesh === null) {
+        if ((toArea.floor.navMesh === null && pathfindingMethod === PathfindingMethod.Standard) || (toArea.floor.voronoiNavMesh === null && pathfindingMethod === PathfindingMethod.Voronoi)) {
             neededToGenerateANavMesh = true;
-            let [genNavMesh, genWalls] = generateNavMesh(toArea.floor);
+            const [genNavMesh, genWalls] = generateNavMesh(toArea.floor, pathfindingMethod);
+            const data: { [key: string]: object } = {
+                walls: JSON.parse(JSON.stringify(genWalls))
+            }
+            pathfindingMethod === PathfindingMethod.Standard ?
+                data.navMesh = JSON.parse(JSON.stringify(genNavMesh)) :
```

Comment:
Same as above, if this can be a nullable value, you can add util method for all `JSON.parse(JSON.stringify(` And do extra try catch logic there to avoid unexpected bugs